src/Icicle/Sea/FromAvalanche/Program.hs

```diff
@@ -205,7 +205,7 @@ seaOfXValue v t
       | ArrayT t' <- t
       -> let len = length vs
              writes arr (v',i)
-              = prim (PrimUpdate $ PrimUpdateArrayPut t')
+              = prim (PrimArray $ PrimArrayPutImmutable t')
```

Comment:
hm- this should be dead code now, because this transform is elsewhere. can you change it to error if an array value makes it this far?